### PR TITLE
Make NewState Parameters non-optional

### PR DESCRIPTION
**Pull request by pattyshack**
More prep work for plumbing fully initialized parameters from block computer
to transaction invoker.

### CODE REVIEW

File: fvm/state/state.go [hunk 1]
```diff
@@ -28,31 +28,53 @@ type mapKey struct {
...
-		updatedAddresses:      make(map[flow.Address]struct{}),
-		updateSize:            make(map[mapKey]uint64),
-		maxKeySizeAllowed:     DefaultMaxKeySize,
-		maxValueSizeAllowed:   DefaultMaxValueSize,
-		maxInteractionAllowed: DefaultMaxInteractionSize,
+type StateParameters struct {
+	// TODO(patrick): nest meter parameters into this
+	MaxKeySizeAllowed     uint64
+	MaxValueSizeAllowed   uint64
+	MaxInteractionAllowed uint64
```
Comment (SaveTheRbtz) [inline fvm/state/state.go:46]: Any reason for them to be public?

File: fvm/fvm.go [hunk 2]
```diff
@@ -51,9 +51,12 @@ func (vm *VirtualMachine) Run(ctx Context, proc Procedure, v state.View, program
...
 			uint(proc.ComputationLimit(ctx)),
 			uint(proc.MemoryLimit(ctx))),
-		state.WithMaxKeySizeAllowed(ctx.MaxStateKeySize),
-		state.WithMaxValueSizeAllowed(ctx.MaxStateValueSize),
-		state.WithMaxInteractionSizeAllowed(ctx.MaxStateInteractionSize))
+		state.StateParameters{
+			MaxKeySizeAllowed:     ctx.MaxStateKeySize,
+			MaxValueSizeAllowed:   ctx.MaxStateValueSize,
+			MaxInteractionAllowed: ctx.MaxStateInteractionSize,
+		},
```
Comment (SaveTheRbtz) [inline fvm/fvm.go:58]: maybe still use `state.DefaultParameters().WithXXX` here to future-proof against additions of new parameters?

**Review by SaveTheRbtz — APPROVED** [review]

File: fvm/fvm.go (hunk 2 above)
Comment (SaveTheRbtz) [reply fvm/fvm.go:58]: Or mabe have a NewStateParameters function so that when a new state parameter is added this code fails to compile?

**Comment by pattyshack** [comment]
converted param fields to private

**Review by janezpodhostnik — APPROVED** [review]

**Comment by pattyshack** [comment]
bors merge

**Comment by pattyshack** [comment]
bors retry

**Comment by pattyshack** [comment]
bors merge

**Comment by pattyshack** [comment]
bors merge

**Pull request merged**